src/ray/common/ray_object.h

```diff
@@ -92,12 +98,19 @@ class RayObject {
   /// large to return directly as part of a gRPC response).
   bool IsInPlasmaError() const;
 
+  /// The node that this object is pinned at.
+  const absl::optional<ClientID> &PinnedAtRayletId() const;
+
  private:
   std::shared_ptr<Buffer> data_;
   std::shared_ptr<Buffer> metadata_;
   const std::vector<ObjectID> nested_ids_;
   /// Whether this class holds a data copy.
   bool has_data_copy_;
+  // If this object is stored in plasma, and reference counting is enabled,
+  // then some raylet must be pinning the object value. This is the address of
+  // that raylet.
+  const absl::optional<ClientID> pinned_at_raylet_id_;
```

Comment:
Since we're already use `meta` to indicate if the object is in plasma, should we reuse `data` for the optional `raylet id` in this case?

Reply:
Hmm we could, but I don't think this is such a good idea because the raylet ID is a pretty different concept from the object value. I'd rather just keep the concepts separate so it's clearer which field to check.